src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -0,0 +1,557 @@
+    // Arguments:
+    // - keys: the key chord of the command to search for
+    // Return Value:
+    // - the command with the given key chord
+    // - nullptr if the key chord is explicitly unbound
+    Model::Command ActionMap::GetActionByKeyChord(Control::KeyChord const& keys) const
+    {
+        // Check the current layer
+        const auto& cmd{ _GetActionByKeyChordInternal(keys) };
+        if (cmd.has_value())
+        {
+            return *cmd;
+        }
+
+        // This key chord is not explicitly bound
+        return nullptr;
+    }
+
+    // Method Description:
+    // - Retrieves the assigned command with the given key chord.
+    // - Can return nullopt to differentiate explicit unbinding vs lack of binding.
+    // Arguments:
+    // - keys: the key chord of the command to search for
+    // Return Value:
+    // - the command with the given key chord
+    // - nullptr if the key chord is explicitly unbound
+    // - nullopt if it was not bound in this layer
+    std::optional<Model::Command> ActionMap::_GetActionByKeyChordInternal(Control::KeyChord const& keys) const
+    {
+        // Check the current layer
+        const auto actionIDPair{ _KeyMap.find(keys) };
+        if (actionIDPair != _KeyMap.end())
+        {
+            // the command was explicitly bound,
+            // return what we found (invalid commands exposed as nullptr)
+            return _GetActionByID(actionIDPair->second);
+        }
+
+        // the command was not bound in this layer,
+        // ask my parents
+        FAIL_FAST_IF(_parents.size() > 1);
+        for (const auto& parent : _parents)
+        {
+            const auto& inheritedCmd{ parent->_GetActionByKeyChordInternal(keys) };
+            if (inheritedCmd.has_value())
+            {
+                return *inheritedCmd;
+            }
+        }
+
+        // This action is not explicitly bound
+        return std::nullopt;
+    }
+
+    // Method Description:
+    // - Retrieves the key chord for the provided action
+    // Arguments:
+    // - action: the shortcut action (an action type) we're looking for
+    // Return Value:
+    // - the key chord that executes the given action
+    // - nullptr if the action is not bound to a key chord
+    Control::KeyChord ActionMap::GetKeyBindingForAction(ShortcutAction const& action) const
+    {
+        return GetKeyBindingForAction(action, nullptr);
+    }
+
+    // Method Description:
+    // - Retrieves the key chord for the provided action
+    // Arguments:
+    // - action: the shortcut action (an action type) we're looking for
+    // - myArgs: the action args for the action we're looking for
+    // Return Value:
+    // - the key chord that executes the given action
+    // - nullptr if the action is not bound to a key chord
+    Control::KeyChord ActionMap::GetKeyBindingForAction(ShortcutAction const& myAction, IActionArgs const& myArgs) const

```

Comment:
IDL actually hates (technically) function overloads. This is fine, but just a caveat. If you want to disambiguate for IDL's sake, you can use "GetKeyBindingForActionWithArgs"

Reply:
> Similarly, for overloaded methods, for the first method in a series of overloads (in lexical order), the compiler uses the original method name for the equivalent ABI interface method. Subsequent overloads are made unique by adding to the original name the smallest unused integer numeral suffix (increasing from 2).


If you really want, you can put `[method_name("GetKeyBindingForActionWithArgs")] blahblah GetKeyBindingForAction(Action, Args);` in the **idl** file, and it will internally be named GKBFAWA but WinRT consumers will see GKBFA